### PR TITLE
DP-107 Removed incorrect class extension from Exchange and MessageQueue

### PR DESCRIPTION
**Pull request by clicktravel-steffan**

### CODE REVIEW

File: cheddar/cheddar-messaging/src/main/java/com/clicktravel/cheddar/infrastructure/messaging/MessagePublisherImpl.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+package com.clicktravel.cheddar.infrastructure.messaging;
+
+import com.clicktravel.cheddar.infrastructure.messaging.exception.MessagePublishException;
+
+/**
+ * A simple message publisher implementation which publishes all messages using an exchange
+ * @param <T> Type of messages published
+ */
+public class MessagePublisherImpl<T extends Message> implements MessagePublisher<T> {
```
Comment (clicktravel-basharat) [inline cheddar/cheddar-messaging/src/main/java/com/clicktravel/cheddar/infrastructure/messaging/MessagePublisherImpl.java]: MessagePublisherImpl -> SimpleMessagePublisher ?

File: cheddar/cheddar-messaging/src/main/java/com/clicktravel/cheddar/infrastructure/messaging/MessageSenderImpl.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ */
+package com.clicktravel.cheddar.infrastructure.messaging;
+
+import com.clicktravel.cheddar.infrastructure.messaging.exception.MessageSendException;
+
+/**
+ * A simple message sender implementation which sends all messages to a queue
+ * @param <T> Type of messages sent
+ */
+public class MessageSenderImpl<T extends Message> implements MessageSender<T> {
```
Comment (clicktravel-basharat) [inline cheddar/cheddar-messaging/src/main/java/com/clicktravel/cheddar/infrastructure/messaging/MessageSenderImpl.java]: MessageSenderImpl -> SimpleMessageSender ?

**Comment by clicktravel-basharat** [comment]
Review done: a couple of renames. "+Impl" is not a standard we've adopted.

File: cheddar/cheddar-messaging/src/main/java/com/clicktravel/cheddar/infrastructure/messaging/MessagePublisherImpl.java (hunk 1 above)
Comment (clicktravel-steffan) [inline cheddar/cheddar-messaging/src/main/java/com/clicktravel/cheddar/infrastructure/messaging/MessagePublisherImpl.java]: I struggled with the choice of name. I decided against prefixing MessagePublisher as xxxMessage normally indicates which type of message is involved i.e. BasicMessage or TypedMessage.

File: cheddar/cheddar-messaging/src/main/java/com/clicktravel/cheddar/infrastructure/messaging/MessageSenderImpl.java (hunk 2 above)
Comment (clicktravel-steffan) [inline cheddar/cheddar-messaging/src/main/java/com/clicktravel/cheddar/infrastructure/messaging/MessageSenderImpl.java]: Same naming issue as mentioned above.

**Comment by clicktravel-basharat** [comment]
OK this is acceptable, given the reasons.

**Pull request merged**